src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6831,8 +6835,13 @@ init_myhook(void)
 
 	init_fd_dup2_list();
 
-	if (compatible_mode)
-		check_exe_sh_bash();
+	/* Need to check whether current process is bash or not under regular & compatible modes.*/
+	check_exe_sh_bash();
+	if (is_bash)
+		/* Disable directory caching inside bash. bash could remove a dir then recreate
+		 * it which causes cache inconsistency. Observed such issue in "configure" in ucx.
+		 */
+		dcache_rec_timeout = 0;
```

Comment:
TBH im not sure how this was working before then with the first version of dir caching. im wondering if we have introduced a new bug or not.

Reply:
I think It is not a bug although I do not understand why I did not encounter previously.